tests/lit-tests/llvm_ident.ispc

```diff
@@ -3,7 +3,7 @@
 
 // Check that version string embeded in !llvm.ident metadata is formed correctly.
 
-// CHECK: Intel(r) Implicit SPMD Program Compiler (Intel(r) ISPC), {{1\.[[:digit:]]+\.[[:digit:]]+(dev)?}} (build commit {{[[:xdigit:]]+}} @ {{([[:digit:]]{8})}}, LLVM [[LLVM_VERSION:[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+]])
```

Comment:
Could you explain what is going on here and why?

Reply:
I don’t know because I can’t understand the code, but I wouldn’t be surprised that this is related to the failure I’ve been seeing for ages in https://github.com/ispc/ispc/issues/2325.

Reply:
Confirmed that this fixes #2325.

Reply:
It pattern matches versions of ISPC and LLVM embedded in bitcode file. LLVM version string is slightly different across different LLVM builds.
E.g., Fedora
```
!0 = !{!"Intel(r) Implicit SPMD Program Compiler (Intel(r) ISPC), 1.19.0dev (build commit 5cce243d430fc30c @ 20230114, LLVM 14.0.5)"}
!1 = !{!"LLVM version 14.0.5 (Fedora 14.0.5-2.fc36)"}
```
Arch
```
!0 = !{!"Intel(r) Implicit SPMD Program Compiler (Intel(r) ISPC), 1.18.1 (build  @ 20221112, LLVM 14.0.6)"}
!1 = !{!"LLVM version 14.0.6"}
```
Our build
```
!0 = !{!"Intel(r) Implicit SPMD Program Compiler (Intel(r) ISPC), 1.19.0dev (build commit 6efb053cacaa69e3 @ 20230111, LLVM 13.0.1)"}
!1 = !{!"LLVM version 13.0.1 (https://github.com/llvm/llvm-project.git 75e33f71c2dae584b13a7d1186ae0a038ba98838)"}
```

Reply:
Changes in the regexp looks good.